transport/tchannel/transport.go

```diff
@@ -333,22 +338,44 @@ func (t *Transport) onPeerStatusChanged(tp *tchannel.Peer) {
 // CreateTLSOutboundChannel creates a outbound channel for managing tls
 // connections with the given tls config and destination name.
 // Usage:
-// 	tr, _ := tchannel.NewTransport(...)
-//  outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
-//  outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
+//
+//		tr, _ := tchannel.NewTransport(...)
+//	 outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
+//	 outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
 func (t *Transport) CreateTLSOutboundChannel(tlsConfig *tls.Config, destinationName string) (peer.Transport, error) {
+	dialerFunc := t.dialer
+	if t.enableMPTCP {
+		dialerFunc = dialMPTCPContext
+	}
 	params := dialer.Params{
 		Config:        tlsConfig,
 		Meter:         t.meter,
 		Logger:        t.logger,
 		ServiceName:   t.name,
 		TransportName: TransportName,
 		Dest:          destinationName,
-		Dialer:        t.dialer,
+		Dialer:        dialerFunc,
 	}
 	return t.createOutboundChannel(dialer.NewTLSDialer(params).DialContext)
 }
 
+// CreateMPTCPOutboundChannel creates a outbound channel for MPTCP
```

Comment:
do we really need this func? I saw only one reference and we can just directly call `t.createOutboundChannel(dialMPTCPContext)` from there? The less exported function, the better

Reply:
`CreateTLSOutboundChannel` have only one reference as well, so I just followed.
Besides, we need to make it as an exported function for testing. Otherwise, we have to make `createOutboundChannel` exported for testing and there will be many code changes. Or we have to delete the unit test for `t.createOutboundChannel(dialMPTCPContext)`